lib/net/ssh/transport/algorithms.rb

```diff
@@ -213,22 +213,33 @@ def prepare_preferred_algorithms!
         options[:compression] = %w(zlib@openssh.com zlib) if options[:compression] == true
 
         ALGORITHMS.each do |algorithm, list|
-          algorithms[algorithm] = list.dup
-
           # apply the preferred algorithm order, if any
           if options[algorithm]
-            algorithms[algorithm] = Array(options[algorithm]).compact.uniq
+            option = Array(options[algorithm]).compact.uniq
+
+            if option.first.start_with?('+')
```

Comment:
I think we're good here: If the option is `nil`, we're not entering that section at all. If it's something else, `Array()` will always make sure we're dealing with an array.